src/method.c

```diff
@@ -215,6 +215,28 @@ static jl_value_t *resolve_globals(jl_value_t *expr, jl_module_t *module, jl_sve
             if (e->head == jl_method_sym || e->head == jl_module_sym || e->head == jl_throw_undef_if_not_sym) {
                 i++;
             }
+            if (e->head == jl_assign_sym && binding_effects) {
+                jl_value_t *lhs = jl_exprarg(e, 0);
+                if (jl_is_globalref(lhs) || jl_is_symbol(lhs)) {
+                    jl_module_t *mod = jl_is_globalref(lhs) ? jl_globalref_mod(lhs) : module;
+                    jl_sym_t *name = jl_is_globalref(lhs) ? jl_globalref_name(lhs) : (jl_sym_t*)lhs;
+                    if (mod == module) {
+                        // Assignment does not create bindings in foreign modules (#54678)
+                        jl_binding_t *b = jl_get_module_binding(mod, name, 1);
+                        jl_binding_partition_t *bpart = jl_get_binding_partition(b, jl_current_task->world_age);
```

Comment:
It may be unlikely that `jl_current_task->world_age` is correct here, since this runs in the toplevel context (constantly changing worlds) not in a fixed world

Reply:
This entire code section needs to be different for actual partition - in this PR, the value is unused.